internal/osutils/autostart/autostart_linux.go

```diff
@@ -5,76 +5,32 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/ActiveState/cli/internal/assets"
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/osutils"
-	"github.com/ActiveState/cli/internal/osutils/shortcut"
 	"github.com/ActiveState/cli/internal/osutils/user"
 	"github.com/ActiveState/cli/internal/subshell/sscommon"
 )
 
 const (
-	autostartDir  = ".config/autostart"
 	autostartFile = ".profile"
 )
 
-func (a *app) enable() error {
-	enabled, err := a.IsEnabled()
-	if err != nil {
-		return errs.Wrap(err, "Could not check if app autostart is enabled")
-	}
-	if enabled {
-		return nil
+func (a *App) enable() error {
+	if err := legacyDisableOnDesktop(a.options.LaunchFileName); err != nil {
+		logging.Error("Cannot properly disable autostart (desktop): %v", err)
```

Comment:
You can error out for real on this. We need to get more comfortable with throwing errors at users so we can actually fix them rather than quietly ignore them to have them bubble up in different ways.